diff --git a/tesseract_monitoring/src/environment_monitor_interface.cpp b/tesseract_monitoring/src/environment_monitor_interface.cpp
index 81526290..54c5d77c 100644
--- a/tesseract_monitoring/src/environment_monitor_interface.cpp
+++ b/tesseract_monitoring/src/environment_monitor_interface.cpp
@@ -360,6 +360,16 @@ bool ROSEnvironmentMonitorInterface::setEnvironmentState(const std::string& monit
   tesseract_rosutils::toMsg(command.floating_joint_states, floating_joints);
   return sendCommands(monitor_namespace, { command });
 }
 
+bool ROSEnvironmentMonitorInterface::setEnvironmentState(
+    const std::string& monitor_namespace,
+    const tesseract_common::TransformMap& floating_joints) const
+{
+  tesseract_msgs::msg::EnvironmentCommand command;
+  command.command = tesseract_msgs::msg::EnvironmentCommand::UPDATE_JOINT_STATE;
+  tesseract_rosutils::toMsg(command.floating_joint_states, floating_joints);
+  return sendCommands(monitor_namespace, { command });
+}
+
 std::vector<std::string>
 ROSEnvironmentMonitorInterface::setEnvironmentState(const std::unordered_map<std::string, double>& joints,
                                                     const tesseract_common::TransformMap& floating_joints) const
 {
